src/Storage/Storage/Common/AzureStorageTable.cs

```diff
@@ -37,14 +42,67 @@ public class AzureStorageTable : AzureStorageBase
         public Uri Uri { get; private set; }
 
         /// <summary>
-        /// Azure storage table constructor
+        /// Constructs AzureStorageTable from track 1 CloudTable.
+        /// Internally it constructs track 2 TableClient used by data plane cmdlets.
+        /// </summary>
+        /// <param name="table">Cloud table object.</param>
+        /// <param name="storageContext">Storage context containing account information used to construct TableClient.</param>
+        /// <param name="options">Table client options which should contain powershell user agent.</param>
+        public AzureStorageTable(CloudTable table, AzureStorageContext storageContext, TableClientOptions options)
+        {
+            this.CloudTable = table;
+            this.Name = table.Name;
+            this.Uri = table.Uri;
+            
+            this.Context = storageContext;
+
+            // construct track 2 table client from track 1 cloud table instance
+            this.TableClient = AzureStorageTable.ConstructTableClientFromCloudTable(this.CloudTable, options);
+        }
+
+        /// <summary>
+        /// Constructs AzureSTorageTable from track 2 TableClient.
+        /// Internally it construcs track 1 CloudTable for property display purpose only.
         /// </summary>
-        /// <param name="table">Cloud table object</param>
-        public AzureStorageTable(CloudTable table)
+        /// <param name="tableClient"></param>
+        public AzureStorageTable(TableClient tableClient)
+        {
+            this.TableClient = TableClient;
```

Comment:
Should be `this.TableClient = tableClient`, or the output table client will always be null.

Normally this should be found in test, if run any request with the output table Client object.
We should do more test on the changed code.